### ISSUE TITLE
Make Number of Days Requested mandatory on Info Tabs (NN)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1525]
### As a ROW Permit Analyst, I need the folder info 'Number of Days Requested' to be a mandatory field, both in back-office and on the portal application. 

**Stakeholders:** Paloma, Kim
**DTS Manager:** Nate

**Acceptance Criteria:**
- Applicable Folders: RW, Temporary Right of Way Use Permit subtype
- The folder info 'Number of Days Requested' is mandatory
- The info is **not mandatory** on the portal application (currently already happening)

**Comment by amenity** [comment]
Make "Number of Days Requested" mandatory. Currently it has a flag, but it is not pink and is possible to issue permit without info. 

We need to know the window they are requesting to work so that we can charge them correctly. 

- It exists on RW permit folder in two subtypes
    - Parking Permits
    - Temporary Use of ROW
    -  Info Tab of TURP folders

Note: 
- Field doesn't exist on EX or DS

**Comment by amenity** [comment]
Might make sense to do along with https://github.com/cityofaustin/atd-data-tech/issues/1918

**Comment by amenity** [comment]
This would be a quick change for CTM.

**Comment by Nadin-Nader** [comment]
Logged in to TEST and went to the Admin side> Folder> Folder and Defaults> RW> Selected the "Setup Folder Defaults" button>Went to the info tab and sorted for the field necessary. I also included the default Procedure for Parking Spaces Permitting.

Folder: RW
SubType: Parking Space
Info: 75390 Number of Days Requested
InfoGroup: Event Start/End
Default Procedure: Parking Spaces Permitting DefaultProcess_RW_00050665

Created Request for CTM (SCTASK0066564):
CALL0032682
REQ0048070
RITM0054358
SCTASK0066564

**Comment by TracyLinder** [comment]
@Nadin-Nader can you update this one also? Did CTM make the changes?

**Comment by mehefkod** [comment]
Status update:  SCTASK0066564 is in Tango team Backlog. We currently only have one developer for 16 hours a week, and will pull this in as soon as we can.

**Comment by Nadin-Nader** [comment]
Status:

Still waiting on CTM Tango to do

**Comment by TracyLinder** [comment]
I just updated the acceptance criteria; moving this to the backlog. It's a good candidate for The Jets, versus waiting on CTM.

**Comment by TracyLinder** [comment]
@Nadin-Nader I'm assigning this to LIT for sprint 31.

**Comment by Nadin-Nader** [comment]
**Status:**
Waiting on Varun Etakula from CTM to deploy

**Comment by TracyLinder** [comment]
@Nadin-Nader what environment is this being deployed to?

**Comment by Nadin-Nader** [comment]
@TracyLinder this is going to TEST and should be done early next week for UAT

**Comment by Nadin-Nader** [comment]
Spoke to Varun on Friday and he explained to me that while e was getting ready to deploy this he found that one of the infos was a bit more complicated to make mandatory;

Folder: RW
SubType: Temporary Right Of Way Use Permit 50500
Info: 75390 Number of Days Requested
InfoGroup: Event Start/End
Default Procedure: TURP Permit Intake DefaultProcess_RW_00050515

The no.of days requested field for Temporary  ROW subtype  is getting the value from the Total Days of Permit Validity which is defaulted to 2 if you don't enter anything. If we enter any value in the no.of days field the total days of permit validity is populated based on the same value, both fields are tightly coupled with each other.

**Comment by Nadin-Nader** [comment]
Sent email to Kim and Paloma, waiting on Response:

Good Morning,

Hope you are both doing well. I am reaching out regarding request 1525, Make Number of Days Requested Mandatory on Info Tabs. I wanted to update you on something the developer came across for this one.

He informed me that the no. of days requested field for Temporary ROW subtype is getting the value from the Total Days of Permit Validity which is defaulted to 2 if you don't enter anything. If we enter any value in the no. of days field the total days of permit validity is populated based on the same value, both fields are tightly coupled with each other.

With that being said do you still want this field mandatory? If so, are you ok with also making the no. of days field the total days of permit validity required?

Please let me know how you would like to proceed as soon as you can so I can get back to the developer and have him complete this request.

Respectfully,

Nadin A. Nader 
IT Business Systems Analyst
Data and Technology Services 
Austin Transportation Department

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader  - Launch IT is not working on this Card Currently and it is on hold

**Comment by TracyLinder** [comment]
The acceptance criteria has been updated. Highlighted below:

![image](https://user-images.githubusercontent.com/49930868/92280166-7574b080-eebe-11ea-9ad2-955e571c0bbf.png)

**Comment by TracyLinder** [comment]
@Nadin-Nader let's discuss this soon. Just talked to Paloma and Kim, and it looks like this functionality already exists in all environments.

**Comment by Nadin-Nader** [comment]
Set meeting with Tracy for tomorrow 9/15/2020 at 11am

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder 
Email from Paloma on 9/1/2020:

Good afternoon all,

I might still be a little confused but I’ll take a stab at answering the questions anyways:
-	Yes, we would still like to proceed with making the “Number of Days Requested” field mandatory.
-	Yes, we’re okay with the default being set at 0 days for this field.
-	The “Total Days of Permit Validity” is not an editable field for us, instead it auto-populates.  I believe this field pulls from “Number of Days Requested” and “Number of Days for Extension” (and, if applicable, sums together).  Given this, it seems that if/when the “Number of Days Requested” field is mandatory (and therefore has a value entered), then by extension, the “Total Days of Permit Validity” will also have a value auto-entered by the system.
-	The default of the “Total Days of Permit Validity” 2 days strikes me as odd, particularly since on many many permits that value is reflected as 0.  Example screenshot below.

One thing I do need to note is that our ask is to make the “Number of Days Requested” field mandatory for staff to enter, NOT for customers to enter (via the ABC Portal). 

Permit #20-096895
 ![Number Days Requested.jpg](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/38980da0-a28c-4547-98cc-bce38a12ab18)

Thanks all!
P

**Comment by Nadin-Nader** [comment]
I Went ahead and cancelled SCTASK0066564  that was with CTM

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder, @Nadin-Nader - Aishwarya will start working on this when #2633 is completed.

**Comment by TracyLinder** [comment]
When I tested this w/ stakeholders on the 4th, this functionality was already working as needed in Prod.
